src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -48694,6 +48694,991 @@ public void M1(C? c, object? x)
+                // (19,15): warning CS8602: Dereference of a possibly null reference.
+                //             : obj.ToString(); // 3
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "obj").WithLocation(19, 15),
+                // (25,15): warning CS8602: Dereference of a possibly null reference.
+                //             ? obj.ToString() // 4
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "obj").WithLocation(25, 15),
+                // (26,15): warning CS8602: Dereference of a possibly null reference.
+                //             : obj.ToString(); // 5
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "obj").WithLocation(26, 15),
+                // (32,15): warning CS8602: Dereference of a possibly null reference.
+                //             ? obj.ToString() // 6
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "obj").WithLocation(32, 15),
+                // (40,15): warning CS8602: Dereference of a possibly null reference.
+                //             : obj.ToString(); // 7
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "obj").WithLocation(40, 15),
+                // (46,15): warning CS8602: Dereference of a possibly null reference.
+                //             ? obj.ToString() // 8
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "obj").WithLocation(46, 15),
+                // (47,15): warning CS8602: Dereference of a possibly null reference.
+                //             : obj.ToString(); // 9
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "obj").WithLocation(47, 15)
+                );
+        }
+
+        [Fact]
+        public void IsPattern_LeftConditionalState()
+        {
+            var source = @"
+class C
+{
+    void M(object? obj)
+    {
+        _ = (obj != null) is true
+            ? obj.ToString()
+            : obj.ToString(); // 1
+    }
+}";
+            var comp = CreateNullableCompilation(source);
+            comp.VerifyDiagnostics(
+                // (8,15): warning CS8602: Dereference of a possibly null reference.
+                //             : obj.ToString(); // 1
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "obj").WithLocation(8, 15));
+        }
+
+        [Fact, WorkItem(53308, "https://github.com/dotnet/roslyn/issues/53308")]
+        public void IsPattern_LeftCondAccess_PropertyPattern_01()
+        {
+            var source = @"
+class Program
+{
+    void M1(B? b)
+    {
+        if (b is { C: { Prop: { } } })
+        {
+            b.C.Prop.ToString();
+        }
+    }
+
+    void M2(B? b)
+    {
+        if (b?.C is { Prop: { } })
+        {
+            b.C.Prop.ToString(); // 1
+        }
+    }
+}
+
+class B
+{
+    public C? C { get; set; }
+}
+
+class C
+{
+    public string? Prop { get; set; }
+}";
+            // Ideally we would not issue diagnostic (1).
```

Comment:
I thought this scenario might just fall out of the change to `learnFromNonNullTest`. Can you clarify what's missing?

Reply:
I didn't debug in a great amount of detail, but I suspect that when we create a slot for `Prop` we don't understand that its containing slot is the slot for `b.C`. Instead we either use some dag temp slot without relating it to the original input, or we create some slot that's accidentally linked to the conditional receiver (left side of `?.C`).